specs/_template/01_main.rst

```diff
@@ -0,0 +1,253 @@
+    It links to the epic ticket number, and provides document metadata
+
+.. Template information:
+    SPECIFICATION_NAME = name of the specification (title)
+    EPIC = epic issue number
+    AUTHOR = your name
+    CREATED_AT = YYYY-MM-dd
+
+.. Please replace the strings below with the values that you want to put in in all of the rst files of the specification.
+
+=======  ==================  ======  ==========
+ epic          title         author   created
+=======  ==================  ======  ==========
+`EPIC`_  SPECIFICATION_NAME  AUTHOR  CREATED_AT
+=======  ==================  ======  ==========
+
+.. _EPIC: gnosis/safe#EPIC
+
+.. _Main:
+
+==================
+SPECIFICATION_NAME
+==================
+
+1. `Main`_
+2. User Interface
+
+   1. `Android`_
+   2. `Chrome`_
+   3. `iOS`_
+
+3. `External Communication`_
+4. Other_
+
+
+.. contents:: Table of Contents
+
+1. Problem Definition
+---------------------
+
+* List your problem definition here
+
+2. Inputs
+-----------
+
+What inputs are needed to solve the problem? 
+
+* User input
+* Network-fetched information
+* Existing information in the database
+
+3. Outputs
+------------
+
+What is output from the functionality that is specified?
+
+* New database data
+* Error messages
+* Some side-effect to other systems
+
+4. Use Case Scenarios
+-----------------------
+
+You can put general information or context for all use case scenarios here.
+Feel free to remove the use case explanation below.
+
+Every use case scenario consists of preconditions – prerequisites
+and assumptions, steps - numbered sequence of steps of the scenario,
+from the user's perspective, and postconditions - results and
+assumptions that must hold after the steps finished.
+
+A use case can *inherit* another use case. That means that
+inheriting use case takes all of the preconditions, steps, and postconditions
+from the inhertied use case.
+
+The inheriting (child) use case
+can override (replace) any step or precondition in the parent
+use case. The overriden steps in child should match the step number
+in parent. The child use case may introduce sub-steps, in that case
+numbering is starts with dot, for example "3.1", "3.2" gives more
+details about step "3" of a parent use case.
+
+4.1. Happy Case
+~~~~~~~~~~~~~~~~~
+
+.. _happy_case_preconditions:
+
+Preconditions
++++++++++++++
+
+* Things that are expected to be true and required for the steps below
+* This are the things that needed for the functionality to work properly
+
+.. _happy_case_steps:
+
+Steps
++++++
+
+1. <Open something>
+
+2. Select "X" action. Something opens.
+
+.. you can reference some steps in other scenarios, when needed:
+
+.. _happy_case_step_3:
+
+3. Type in "abcd". and so on
+
+
+.. _happy_case_postconditions:
+
+Postconditions
+++++++++++++++
+
+* Changes to the database, to user interface
+* Changes to the backend systems' databases
+* Changes in the smart contracts and so on.
+
+
+4.2. Edge Case Scenario
```

Comment:
Can we find a better name than edge case scenario?

Most of the time this describes error cases, which are not really edge cases.
Edges cases sound for me like something that should not happen, a little misleading imo.

Reply:
Ahh now that I read the rest I get what you wanted to do here ... that is kind of confusing (it is more a reminder "Think about edge cases", right?)

Reply:
I would not call it edge cases either. More like exceptions to the normal user flow.

Reply:
yes, the intention was to provide a placeholder text. What if I enclose the title into `< ... >`?

Reply:
I'll call it scenario extension